### ISSUE TITLE
PARAMETER statement ambiguous without PROGRAM statement

### ISSUE THREAD

**Issue by pjh40** [lfortran/lfortran#5269]
```f90
  PARAMETER (NUMBER_OF_SENATORS = 100)
END PROGRAM
```
Results in the error
```sh
syntax error: Internal Compiler Error: syntax is ambiguous in the parser
 --> 5250.f90:1:3
  |
1 |   PARAMETER (NUMBER_OF_SENATORS = 100)
  |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 


Note: Please report unclear, confusing or incorrect messages as bugs at
https://github.com/lfortran/lfortran/issues.
```

using `main` branch `0.41.0-224-g11df6a825`.  Adding a program statement allows compilation without error.

**Comment by certik** [comment]
Thanks! I can reproduce it. As a workaround, one can prepend "program something", then it parses.

Without the program statement we parse it in interactive mode, and then when we encounter "end program" we put all the "interactive statements" into a program. This allows us to maintain just one parser behavior.

In interactive mode the parameter is apparently ambiguous:
```
>>> parameter(x = 10)                                              1,18  ]
syntax error: Internal Compiler Error: syntax is ambiguous in the parser
 --> input:1:1
  |
1 | parameter(x = 10)
  | ^^^^^^^^^^^^^^^^^^
```

We have to debug it. It probably parses it either as a call to `function parameter(x)`, or as a `parameter` statement. This can be resolved with priorities, in this case we would give higher priority to the parameter statement, and in interactive mode one would either have to rename a function `parameter` or assign the result to a variable first. Not ideal, but it shouldn't hopefully be a big problem. This might be one of the few cases of a true ambiguity when making Fortran interactive. The non-interactive *standard* grammar is not ambiguous.

It's interesting that it works inside a function / program. This might be related #5273, and maybe the reason why I separated declarations and executable statements in the first place. But in the worst case we can fix it up during semantic analysis.

**Comment by certik** [comment]
The above bug is fixed by #5276. We can worry about interactive mode and calling user defined function `parameter` later, this is a very corner case. The main thing for beta quality is to compile existing codes, which are all non-interactive, and that use case should work now.
